ErgoScript-Developer-Course/batch-0/Arkan/HW2_Ergoscript_Recap.md

```diff
@@ -0,0 +1,76 @@
+### Booleans, if/else, pair
+```scala
+{
+    val whichOutputHasMoreErgs = SELF.R4[Int].get
+    val outValues: (Long, Long) = (OUTPUTS(0).value, OUTPUTS(1).value)
+    val firstHasMore: Boolean = outValues._1 > outValues._2
+    if (whichOutputHasMoreErgs == 0)
+        sigmaProp(firstHasMore)
+    else if (whichOutputHasMoreErgs == 1)
+        sigmaProp(!firstHasMore)
+    else
+        sigmaProp(1==1)
+}
+```
+### Coll, lambda, Long
+```scala
+{
+    val twoInputs: Coll[Box] = Coll(INPUTS(0), INPUTS(1))
+    val values: Coll[Long] = twoInputs.map{
+        (input: Box) =>
+            (input.value)
+	}
+
+    val doubled: Coll[Long] = values.map{
+        (value: Long) =>
+            (value * 2L)
+    }
+    val sum: Long = doubled.fold(0L, {
+    (first: Long, second: Long) =>
```

Comment:
The syntax is correct, but it makes more sense to call the `first` variable the `acc` or `accumulator` since it represents the currently accumulated sum. And `second` would have a better name as `doubleVal` for example. Makes things more clear while reading.

Reply:
Sure, makes sense. Thanks.